### DISCUSSION TITLE
[Bug] [python] Tornado framework link cannot be traced correctly

### DISCUSSION THREAD

**Discussion by YueLangsugar** [apache/skywalking#9133] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Python Agent (apache/skywalking-python)

### What happened

I started my tornado service and sent two identical requests, but only one link?

![image](https://user-images.githubusercontent.com/48346272/170223876-a4efd4a5-4a4b-4a81-a743-8b34ebae0438.png)

### What you expected to happen

There is a problem with the link

### How to reproduce

There is my code.  tornado==6.1

```
import asyncio
from skywalking import agent, config
from tornado.web import Application, RequestHandler, url
from tornado.httpserver import HTTPServer
from tornado.ioloop import IOLoop


class SwTornadoHandler(RequestHandler):

    async def get(self):
        await asyncio.sleep(0.4)
        self.write({"ok": 1})


if __name__ == '__main__':
    app = Application([
        url(r"/api/web/test", SwTornadoHandler)
    ])

    http_server = HTTPServer(app, xheaders=True)
    http_server.bind(9696)
    http_server.start(2)

    config.init(
        protocol="http",
        service_instance=f"127.0.0.1:9696",
        collector_address="127.0.0.1:12800",
        service_name="python-api-service")
    config.disable_plugins = ["sw_falcon", "sw_psycopg2", "sw_pyramid", "sw_sanic"]
    agent.start()

    IOLoop.current().start()



```

### Anything else

In skywalking/trace/context.py . I modified the code and found it normal, but I don't know why.
source code:
```
def _spans_dup():
        spans = __spans.get()[:]
        __spans.set(spans)

        return spans
```

modified code:

```
def _spans_dup():
        # spans = __spans.get()[:]
        # __spans.set(spans)
        # 
        # return spans
        return __spans.get()

```

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Superskyyy** [comment, marked as answer]
@YueLangsugar Thanks for reporting the potential problem, you checked the `Yes I am willing to submit a PR!` button, are you currently investigating a fix? It's not necessarily called a bug as there's no problem with the agent code, it's related to how Python works.

This is the same problem that many issues have mentioned, pre-fork server models do not work out-of-box with spawned threads in Python, that is - you used `http_server.start(2)`. Here you have two identical processes handling the traffic where only one agent is attached to the original process. 

You need to pass a post_fork hook definition to the server, and pass to the server config/ startup command
```
# This is pseudo code
def post_fork():
	agent.start()
	blabla
```
as we discussed in many places. (no one mentioned explicitly Tornado yet, so you could help to enhance the documentation FAQ to mention how to configure the hook for Tornado, I'm not sure if there's a way for its built-in server? It's a good idea to investigate this - for built-in Tornado HTTP server)
https://skywalking.apache.org/docs/skywalking-python/latest/en/setup/faq/how-to-use-with-uwsgi/
https://github.com/apache/skywalking/issues/9071 > on enhancing this experience and make it fully automatic for users.
https://github.com/apache/skywalking/issues/8566#issuecomment-1137451755

Please also reformat your issue properly using ```. It's currently hard to read.

**Comment by YueLangsugar** [comment]
I‘m sorry. tornado version is 5.1.1. not 6.1.
